app/controllers/api/databases.php

```diff
@@ -115,19 +127,69 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
+    } catch (\Exception $e) {
+        $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $collectionId);
+        $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $collection->getInternalId());
+        throw $e;
     }
 
     $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $collectionId);
     $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $collection->getInternalId());
 
+    if ($type === Database::VAR_RELATIONSHIP && $options['twoWay']) {
+        $twoWayKey = $options['twoWayKey'];
+        $options['relatedCollection'] = $collection->getId();
+        $options['twoWayKey'] = $key;
+        $options['side'] = Database::RELATION_SIDE_CHILD;
+
+        try {
+            $twoWayAttribute = new Document([
+                '$id' => ID::custom($db->getInternalId() . '_' . $relatedCollection->getInternalId() . '_' . $twoWayKey),
+                'key' => $twoWayKey,
+                'databaseInternalId' => $db->getInternalId(),
+                'databaseId' => $db->getId(),
+                'collectionInternalId' => $relatedCollection->getInternalId(),
+                'collectionId' => $relatedCollection->getId(),
+                'type' => $type,
+                'status' => 'processing', // processing, available, failed, deleting, stuck
+                'size' => $size,
+                'required' => $required,
+                'signed' => $signed,
+                'default' => $default,
+                'array' => $array,
+                'format' => $format,
+                'formatOptions' => $formatOptions,
+                'filters' => $filters,
+                'options' => $options,
+            ]);
+
+            $dbForProject->checkAttribute($relatedCollection, $twoWayAttribute);
+            $dbForProject->createDocument('attributes', $twoWayAttribute);
+        } catch (DuplicateException) {
+            $dbForProject->deleteDocument('attributes', $attribute->getId());
+            throw new Exception(Exception::ATTRIBUTE_ALREADY_EXISTS);
+        } catch (LimitException) {
+            $dbForProject->deleteDocument('attributes', $attribute->getId());
+            throw new Exception(Exception::ATTRIBUTE_LIMIT_EXCEEDED, 'Attribute limit exceeded');
+        } catch (\Exception $e) {
+            $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $relatedCollection->getId());
+            $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $relatedCollection->getInternalId());
+            throw $e;
+        }
+
+        $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $relatedCollection->getId());
+        $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $relatedCollection->getInternalId());
```

Comment:
Can we not achieve the same thing by using finally? 
```suggestion
        } finally {
            $dbForProject->deleteCachedDocument('database_' . $db->getInternalId(), $relatedCollection->getId());
            $dbForProject->deleteCachedCollection('database_' . $db->getInternalId() . '_collection_' . $relatedCollection->getInternalId());
        }
```

Reply:
I see 👍  okay